spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -729,8 +729,13 @@ public void testBatchListenerManual() throws Exception {
 				return invocation.callRealMethod();
 			}
 			finally {
+				boolean smallOffsetCommitted = false;
 				for (Entry<TopicPartition, OffsetAndMetadata> entry : map.entrySet()) {
-					if (entry.getValue().offset() == 2) {
+					if (entry.getValue().offset() == 1) {
+						smallOffsetCommitted = true;
+						logger.error("The highest offset should be the only one committed.");
```

Comment:
Well, it's really an `IllegalState` to be here.
According our new logic we we should never get lower offset for the `commit()`.
I'll fix on merge